discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -239,7 +240,11 @@ def ipfs_id_multiaddr(self):
         return self._multiaddr
 
     def cid_is_valid(self, cid):
-        if cid and re.match(r"^Qm[a-zA-Z0-9]{44}$", cid):
-            return True
+        if not cid:
+            return False
 
-        return False
+        try:
+            make_cid(cid)
```

Comment:
is there any value in checking the returned multihash object?

Reply:
it's an object of class cid, not a string you can directly use.  it would break the current code because we assume strings

```
>>> make_cid
<function make_cid at 0x105ca5950>
>>> make_cid('QmREZWkmPYexzV1zUGwuqyFtPJkkkEGBBnS1MohPWFxAxn')
CIDv0(version=0, codec=dag-pb, multihash=b'\x12 +\x03\xb3|I\xa0*\x91^Hk\x99C\x80$\x8a\xef\x87..')
>>> c = make_cid('QmREZWkmPYexzV1zUGwuqyFtPJkkkEGBBnS1MohPWFxAxn')
>>> str(c)
'QmREZWkmPYexzV1zUGwuqyFtPJkkkEGBBnS1MohPWFxAxn'
>>> c
CIDv0(version=0, codec=dag-pb, multihash=b'\x12 +\x03\xb3|I\xa0*\x91^Hk\x99C\x80$\x8a\xef\x87..')
>>> type(c)
<class 'cid.cid.CIDv0'>
```

Reply:
just want to bump this

Reply:
lets leave this as is